### ISSUE TITLE
Bug: Paths - AllowImportNames cause config error

### ISSUE THREAD

**Issue by RalphK66** [eslint/eslint#18704]
### Environment

Node version: v18.20.3
npm version: 10.7.0
Local ESLint version: 8.51.0
Global ESLint version: none
Operating System: macOS Sonoma 14.5 (23F79)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  rules: {
    'no-restricted-imports': [
      'error',
      {
        paths: [
          {
            name: 'someExternalPackage',
            allowImportNames: ['ValidImportOne', 'ValidImportTwo'],
          },
        ],
      },
    ],
  },
};

```
</details>

```js
<!-- Paste your code here -->
```

### What did you expect to happen?

No config errors

### What actually happened?

```
npx eslint --print-config ./.eslintrc.js

Oops! Something went wrong! :(

ESLint: 8.51.0

Error: .eslintrc.js:
        Configuration for rule "no-restricted-imports" is invalid:
        Value {"paths":[{"name":"someExternalPackage","allowImportNames":["ValidImportOne","ValidImportTwo"]}]} should be string.
        Value {"paths":[{"name":"someExternalPackage","allowImportNames":["ValidImportOne","ValidImportTwo"]}]} should NOT have additional properties.
        Value {"paths":[{"name":"someExternalPackage","allowImportNames":["ValidImportOne","ValidImportTwo"]}]} should match some schema in anyOf.
        Value {"name":"someExternalPackage","allowImportNames":["ValidImportOne","ValidImportTwo"]} should be string.
        Value {"name":"someExternalPackage","allowImportNames":["ValidImportOne","ValidImportTwo"]} should NOT have additional properties.
        Value {"name":"someExternalPackage","allowImportNames":["ValidImportOne","ValidImportTwo"]} should match some schema in anyOf.
        Value [{"paths":[{"name":"someExternalPackage","allowImportNames":["ValidImportOne","ValidImportTwo"]}]}] should match some schema in anyOf.
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/vitejs-vite-vopwsl?file=package.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @RalphK66, thanks for the issue!
 
> Local ESLint version: 8.51.0

The `allowImportNames` option was added in ESLint [v9.0.0](https://eslint.org/blog/2024/04/eslint-v9.0.0-released) so you'll need to update your ESLint version to use this option.
